test/integration/targets/tower_group/tasks/main.yml

```diff
@@ -14,7 +14,6 @@
     name: Some Group
     inventory: Demo Inventory
     state: present
-    source: ec2
```

Comment:
Associating the parent groups is a new feature, and we would really like to get some integration test coverage here.

Developing the corresponding assertion might be kind of difficult, since there's no easy way to return what the existing groups are.

I would suggest that maybe you could create a group, create its future parent, and then invoke the tower_group module again, setting the parent, and then assert that the changed status is true. That would at least demonstrate that the group parent functionality is being used. Maybe a reference case where parents are not changed would be good to have as well.

Reply:
I was just thinking about testing. Its difficult that the modules don't support just fetching the existing data about a given object. I'm not sure if that's something that would be desirable or not to implement. 

I'll look at added some tests as you suggested. I'll also revert the commit that removes the `source: ec2` line from the test as that is valid again after reverting the change that removed the source parameter.